folium/folium.py

```diff
@@ -948,6 +950,40 @@ def json_style(style_cnt, line_color, line_weight, line_opacity,
         self.template_vars.setdefault('geo_styles', []).append(style)
         self.template_vars.setdefault('gjson_layers', []).append(layer)
 
+    @iter_obj('image_overlay')
+    def image_overlay(self, data, image_name=None, filename='data.png',
+                      lower_left=[-90,-180], upper_right=[90,180], opacity=0.25):
+        """Simple image overlay of raster data from URL or numpy array.
+
+        """
+
+        if isinstance(data, str):
+            pass
+        try:
+            imageio.imwrite(filename, data)
+        except:
+            raise ValueError('Data is not proper image format (NxM, NxMx3 or NxMx4)')
```

Comment:
I do not know `imageio` very well to know all the exceptions that might occur.  Maybe a patter that catches `Exception as e` and re-raises it might be better than a `ValueError`.  (Bare in mind that I never did this and I am unsure if it works the way I am imagining :wink:)